knative-operator/pkg/webhook/knativeeventing/webhook_validating_test.go

```diff
@@ -1,66 +1,72 @@
-package knativeeventing_test
+package knativeeventing
 
 import (
 	"context"
 	"os"
 	"testing"
 
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis"
-	. "github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/knativeeventing"
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/testutil"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/scheme"
 	eventingv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
 )
 
+var (
+	ke1 = &eventingv1alpha1.KnativeEventing{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ke1",
+		},
+	}
+	ke2 = &eventingv1alpha1.KnativeEventing{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ke2",
+		},
+	}
+
+	decoder types.Decoder
+)
+
 func init() {
 	apis.AddToScheme(scheme.Scheme)
-}
-
-var ke1 = &eventingv1alpha1.KnativeEventing{
-	ObjectMeta: metav1.ObjectMeta{
-		Name: "ke1",
-	},
-}
-var ke2 = &eventingv1alpha1.KnativeEventing{
-	ObjectMeta: metav1.ObjectMeta{
-		Name: "ke2",
-	},
+	decoder, _ = admission.NewDecoder(scheme.Scheme)
 }
 
 func TestInvalidNamespace(t *testing.T) {
 	os.Clearenv()
 	os.Setenv("REQUIRED_EVENTING_NAMESPACE", "knative-eventing")
+
 	validator := Validator{}
-	validator.InjectDecoder(&mockDecoder{ke1})
-	result := validator.Handle(context.TODO(), types.Request{})
+	validator.InjectDecoder(decoder)
```

Comment:
This also returns an error but is always nil, weird...

Reply:
Yeah, we usually just don't care in tests :shrug: